keps/sig-apps/sidecarcontainers.md

```diff
@@ -123,8 +123,33 @@ The proposal can broken down into four key pieces of implementation that all rel
 * Sidecars are terminated after normal containers
 * Sidecars start before normal containers
 
+#### API Changes:
 As this is a change to the Container spec we will be using feature gating, you will be required to explicitly enable this feature on the api server as recommended [here](https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#adding-unstable-features-to-stable-versions).
 
+New field `ContainerLifecycle` will be added to the Container spec:
+
+```go
+type Container struct {
+  //Container Lifecycle.
+  //One of Standard, Sidecar.
+  //Defaults to Standard
+  //+optional
+  ContainerLifecycle ContainerLifecycle `json:"containerLifecycle,omitempty" protobuf:"bytes,22,opt,name=containerLifecycle,casttype=ContainerLifecycle"`
```

Comment:
InitContainers are also using type Container - this is meaningless for them, right?

Reply:
There are use-cases where you want to have a sidecar running during the pod init phase. For example, you might want to run the Envoy proxy as a sidecar during init in case any init containers try to perform network calls and you'd like to shape that network traffic also (in the same way you do with regular containers).